playwright-gh-pages/deploy-report-pages/action.yml

```diff
@@ -0,0 +1,88 @@
+name: 'Publish to GH pages'
+description: 'Publishes test artifacts to GitHub Pages and comments on the PR with the results and links.'
+author: 'frontend@grafana'
+
+inputs:
+  github-token:
+    description: 'Token for the repository. Can be passed in using `{{ secrets.GITHUB_TOKEN }}`.'
+    required: true
+  retention-days:
+    description: 'Number of days to retain the reports. Default is 30.'
+    required: true
+    default: 30
+  pr-comment-summary:
+    description: 'Whether to comment the PR with the test results. Default is true.'
+    required: true
+    default: true
+  artifact-pattern:
+    description: 'Pattern to match the artifacts. Default is "playwright-report-*"'
+    required: true
+    default: 'playwright-report-*'
+  
+runs:
+  using: "composite"
+  steps:
+    - name: Download report artifacts
+      uses: actions/download-artifact@v4
+      with:
+        path: all-reports
+        pattern: ${{ inputs.artifact-pattern }}
+        merge-multiple: true
+
+    - name: Delete any existing comment
+      uses: thollander/actions-comment-pull-request@v3
+      with:
+        comment-tag: test-results
+        mode: delete
+
+    - name: Set a timestamp
+      shell: bash
+      id: timestampid
+      run: echo "timestamp=$(date --utc +%Y%m%d)" >> "$GITHUB_OUTPUT"
+
+    - name: Push the new files to github pages
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ inputs.github-token }}
+        publish_dir: all-reports
+        destination_dir: ${{ steps.timestampid.outputs.timestamp }}/${{ github.event.number }}
```

Comment:
By reading the docs, I believe `github.event.number` is set only for pull requests:

- https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#using-data-from-the-triggering-workflow
- https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request

I think this won't work for other events, such as schedule/cron, which is something we also have in the [docs examples](https://grafana.com/developers/plugin-tools/e2e-test-a-plugin/ci).

Do we want to change this and make it a bit more generic/configurable?

Reply:
Yes good shout! 

Not tested yet, but [this](https://github.com/grafana/plugin-actions/pull/50/commits/6b348c6b82c729bd0c6251ac02c48019f9563c28) is what I came up with. Let me know what you think.

Reply:
Okay tested now. Had to make a few changes so probably best to look at the final diff. Basically, I'm now using the `github.event.number` if it's set, and otherwise `github.run_id` (this I believe should always exist. 

See example run [here](https://github.com/sunker/test-panel/actions/runs/13031354709?pr=1) when a plugin `ci.yml` workflow is triggered after merging to the main branch. Some of the steps such as updating the report comment is ignored in this case. Notice the result table is written to the workflow summary. The report URL in this case is using the run_id rather than the pr number.